python/nav/ipdevpoll/plugins/snmpcheck.py

```diff
@@ -100,13 +92,10 @@ def _dispatch_down_event(self):
     def _dispatch_up_event(self):
         EVENT.end(None, self.netbox.id).save()
 
-
-@transaction.atomic()
-def get_snmp_agent_down_set():
-    """Returns a set of netbox ids where the SNMP agent is known to be down"""
-    infinity = datetime.datetime.max
-    down = AlertHistory.objects.filter(
-        netbox__isnull=False, event_type__id='snmpAgentState',
-        end_time__gte=infinity).values('netbox__id')
-    down_set = set(row['netbox__id'] for row in down)
-    return down_set
+    @transaction.atomic()
+    def _currently_down(self):
+        infinity = datetime.datetime.max
+        return AlertHistory.objects.filter(
```

Comment:
actually, I would suggest using something like
```AlertHistory.objects.unresolved('snmpAgentState').filter(netbox=self.netbox.id).exists()```

I'm just going to commit that myself before merging :)